in_toto/verifylib.py

```diff
@@ -239,16 +239,17 @@ def verify_all_steps_signatures(layout, chain_link_dict):
 
   """
   for step in layout.steps:
-    keys_dict = {}
-
     # Find the according link for this step
     key_link_dict = chain_link_dict[step.name]
 
-    # Create the dictionary of keys for this step
-    for pubkeyid in step.pubkeys:
-      keys_dict[pubkeyid] = layout.keys[pubkeyid]
-
     for keyid, link in six.iteritems(key_link_dict):
+      keys_dict = {}
+
+      # Create the dictionary of keys for this step
+      for pubkeyid in step.pubkeys:
+          if pubkeyid == keyid:
+            keys_dict[pubkeyid] = layout.keys[pubkeyid]
```

Comment:
Here are some notes,
- make sure to always have consistent indentation (2 spaces per level, 4 for line continuation),
- use `if pubkeyid in step.pubkeys` instead of the loop,
- I think it makes sense to fail early (plus give user feedback) , e.g. something like:
```python
if pubkeyid in step.pubkeys:
  keys_dict[pubkeyid] = layout.keys[pubkeyid]
else:
  raise SignatureVerificationError("<some descriptive message here>")
```
- also remove the signature verification part from the [threshold verification function]( https://github.com/in-toto/in-toto/blob/develop/in_toto/verifylib.py#L986-L993) as mentioned in the related [issue](https://github.com/in-toto/in-toto/issues/82).
- try to add some comments that describe what your changes are good for,
- if you fix an issue please mention the issue in the PR.